pkg/alertmanager/alertmanager.go

```diff
@@ -664,9 +653,36 @@ func (am *Alertmanager) buildIntegrationsMap(gCfg *config.GlobalConfig, nc []*de
 		integrationsMap[rcv.Name] = integrations
 	}
 
+	// Template validation shouldn't be dependent on whether receivers exist. So, in case we didn't hot-load any
+	// templates, we load our best guess as to the appropriate one (Grafana vs Cloud) here to ensure the definitions
+	// are valid.
+	// This might appear different from the above dynamic approach that depends on receiver types, and it is, but
+	// currently AMs should not have mixed receiver types. So, this is a safe (and necessary) workaround.
+	if tmpl == nil && gTmpl == nil {
+		_, err := am.loadTemplates(templates, WithCustomFunctions(am.cfg.UserID))
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I'm fine with this workaround for now, but relying on an optional, experimental argument to choose which set of functions we're using to validate templates is not ideal :grimacing:

Reply:
Agreed. Not much else to do though without a much larger refactor. At least it only ever runs if a tenant has templates but no receivers, which is rare given the default receiver needed in the notification policy.